docs/devel/hello-world-gadget.md

```diff
@@ -0,0 +1,514 @@
+```c
+#define NAME_MAX 255
+
+struct event {
+	__u32 pid;
+	__u8 comm[TASK_COMM_LEN];
+	__u8 filename[NAME_MAX];
+};
+```
+
+And create the logic to fill those fiels in the `enter_openat` program.
+
+```c
+	bpf_get_current_comm(&event.comm, sizeof(event.comm));
+	bpf_probe_read_user_str(&event.filename, sizeof(event.filename), (const char *)ctx->args[1]);
+```
+
+Build and run the gadget again. Now it provides more information.
+
+```
+$ sudo -E ig image build -t mygadget:latest .
+....
+$ sudo -E ig run mygadget:latest
+INFO[0000] Experimental features enabled
+RUNTIME.CONTAINERNAME                         PID                      COMM                     FILENAME
+WARN[0000] The gadget doesn't provide metadata
+                                              11305                    Chrome_ChildIOT          /dev/shm/.org.chromium.…
+                                              11305                    ThreadPoolForeg          /home/mvb/.config/Slack…
+                                              11305                    Chrome_ChildIOT          /dev/shm/.org.chromium.…
+                                              11305                    ThreadPoolForeg          /home/mvb/.config/Slack…
+                                              11305                    Chrome_ChildIOT          /dev/shm/.org.chromium.…
+                                              1349                     containerd               /var/lib/containerd/io.…
+                                              1349                     containerd               /var/lib/containerd/io.…
+                                              1349                     containerd               /var/lib/containerd/io.…
+```
+
+## Creating a metadata file
+
+The above formatting is not totally great, the pid column is taking a lot of space while the
+filename is being trimmed. The metadata file contains extra information about the gadget, among
+other things, it can be used to specify the format to be used.
+
+An initial version of the metadata file can be created by passing `--update-metadata` to the build command:
+
+```bash
+$ sudo -E ig image build . -t mygadget --update-metadata
+```
+
+It'll create a `gadget.yaml` file:
+
+```yaml
+name: 'TODO: Fill the gadget name'
+description: 'TODO: Fill the gadget description'
+tracers:
+  events:
+    mapName: events
+    structName: event
+structs:
+  event:
+    fields:
+    - name: pid
+      description: 'TODO: Fill field description'
+      attributes:
+        width: 16
+        alignment: left
+        ellipsis: end
+    - name: comm
+      description: 'TODO: Fill field description'
+      attributes:
+        width: 16
+        alignment: left
+        ellipsis: end
+    - name: filename
+      description: 'TODO: Fill field description'
+      attributes:
+        width: 16
+        alignment: left
+        ellipsis: end
+```
+
+Let's edit the file to customize the output. We define some templates for well-known fields like
+pid, comm, etc.
```

Comment:
I agree. I was even thinking that we could create specific data types for them, so the developer doesn't have to worry about setting the template, but it could be too much coupling...